nscq/src/types.rs

```diff
@@ -0,0 +1,107 @@
+use std::ffi::c_void;
+
+pub type NscqRc = i8;
+
+pub type NscqSession = *mut c_void;
+
+pub type NscqTnvlStatus = i8;
+
+pub type NscqArch = i8;
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct NscqSessionResult {
+    pub rc: NscqRc,
+    pub session: NscqSession,
+}
+
+pub type NscqObserver = *mut c_void;
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct NscqObserverResult {
+    pub rc: NscqRc,
+    pub observer: NscqObserver,
+}
+
+pub const NSCQ_ATTESTATION_REPORT_SIZE: usize = 0x2000;
```

Comment:
Hum, the attestation reports have size `8192` for the GPU devices. Are we sure that for nvswitches it's 2000 ?

Reply:
it's hex value, 0x2000 = 8192, I can change it to decimal if you don't like the round hex number :-)
it's copy pasted from the original